### PR TITLE
Replace alpine images with scratch images

### PR DESCRIPTION
**Pull request by rturner3**
SPIRE is moving away from the alpine Docker release images in favor of scratch images that contain only the release binary to minimize the size of the images and include only the software that is necessary to run in the container.

### CODE REVIEW

**Review by azdagron — DISMISSED** [review]
Very good!

**Comment by dennisgove** [comment]
There was a regression [issue](https://github.com/spiffe/spire/issues/3581) created in `1.5.0` fixed in `1.5.1`. Should all of these be set to `1.5.1` instead?

File: docker-compose/nested-spire/README.md [hunk 1]
```diff
@@ -48,15 +48,15 @@ We define all the services for the tutorial in the [docker-compose.yaml](docker-
...
+       image: ghcr.io/spiffe/spire-server:1.5.1
        hostname: root-server
        volumes:
          - ./root/server:/opt/spire/conf/server
        command: ["-config", "/opt/spire/conf/server/server.conf"]
      root-agent:
        # Share the host pid namespace so this agent can attest the nested servers
        pid: "host"
-       image: gcr.io/spiffe-io/spire-agent:1.0.0
+       image: ghcr.io/spiffe/spire-agent:1.5.1
```
Comment (MarcosDY) [inline docker-compose/nested-spire/README.md:59]: there are some configurations that uses 1.5.1 and another uses 1.5.0, I think we may use 1.5.1 on all places

Comment (rturner3) [reply docker-compose/nested-spire/README.md:59]: Agreed, updated.

**Review by azdagron — APPROVED** [review]

**Comment by omris94** [comment]
Hey there, it seems like you broke the K8s tutorial:
https://stackoverflow.com/questions/74840467/spiffe-error-while-deploying-client-agent-pods/74841373#74841373

**Pull request merged**